src/fides/api/ops/util/consent_util.py

```diff
@@ -48,19 +56,105 @@ def should_opt_in_to_service(
 
     # OLD WORKFLOW
     if privacy_request.consent_preferences:
-        return all(
-            consent_pref["opt_in"]
-            for consent_pref in privacy_request.consent_preferences
+        return (
+            all(
+                consent_pref["opt_in"]
+                for consent_pref in privacy_request.consent_preferences
+            ),
+            [],  # Don't need to return the filtered preferences, this is just relevant for the new workflow
         )
 
     # NEW WORKFLOW
-    filtered_preferences = filter_privacy_preferences_for_propagation(
+    relevant_preferences = filter_privacy_preferences_for_propagation(
         system, privacy_request.privacy_preferences
     )
-    if not filtered_preferences:
-        return None  # We should do nothing here
+    if not relevant_preferences:
+        return None, []  # We should do nothing here
 
-    return all(
-        filtered_pref.preference == UserConsentPreference.opt_in
-        for filtered_pref in filtered_preferences
+    # Collapse relevant preferences into whether we should opt-in or opt-out
+    preference_to_propagate: UserConsentPreference = (
+        UserConsentPreference.opt_out
+        if any(
+            filtered_pref.preference == UserConsentPreference.opt_out
+            for filtered_pref in relevant_preferences
+        )
+        else UserConsentPreference.opt_in
     )
+
+    # Hopefully rare final filtering in case there are conflicting preferences
+    filtered_preferences: List[PrivacyPreferenceHistory] = [
+        pref
+        for pref in relevant_preferences
+        if pref.preference == preference_to_propagate
+    ]
+
+    # Return whether we should opt in, and the filtered preferences so we can update those for consent reporting
+    return preference_to_propagate == UserConsentPreference.opt_in, filtered_preferences
+
+
+def cache_initial_status_and_identities_for_consent_reporting(
+    privacy_request: PrivacyRequest,
+    connection_config: ConnectionConfig,
+    relevant_preferences: List[PrivacyPreferenceHistory],
+    relevant_user_identities: Dict[str, Any],
+) -> None:
+    """Add a pending system status and cache relevant identities on the applicable PrivacyPreferenceHistory
+    records for consent reporting.
+
+    Preferences that aren't relevant for the given system/connector are given a skipped status.
+
+    Typically used when *some* but not all privacy preferences are relevant.  Otherwise,
+    other methods just mark all the preferences as skipped.
+    """
+    for pref in privacy_request.privacy_preferences:
+        if pref in relevant_preferences:
+            pref.update_secondary_user_ids(relevant_user_identities)
+            pref.cache_system_status(
+                connection_config.system_key, ExecutionLogStatus.pending
+            )
+        else:
+            pref.cache_system_status(
+                connection_config.system_key, ExecutionLogStatus.skipped
+            )
```

Comment:
Method shared between saas and email consent connectors to add a mix of "skipped" and "pending" logs for the case where only a subset of the privacy preferences on a privacy request are relevant for the given connector.